packages/main/src/plugin/docker/docker-compatibility.ts

```diff
@@ -69,14 +69,17 @@ export class DockerCompatibility {
   }
 
   protected getTypeFromServerInfo(
-    info: { OperatingSystem?: string },
+    info: { Name?: string; OperatingSystem?: string },
     podmanInfo?: unknown,
   ): DockerSocketServerInfoType {
     if (info.OperatingSystem === 'Docker Desktop') {
       return 'docker';
     } else if (info.OperatingSystem === 'podman' || podmanInfo) {
       // if podman info is available, then it is podman
       return 'podman';
+    } else if (isLinux() && info.Name === getHostname()) {
+      // Docker Engine
+      return 'docker';
```

Comment:
question: hello, I don't see how this part can be specific to docker and not the others. IMHO relying only on the hostname is too weak.

Reply:
It just assumes that if there is a local docker.sock that is not podman, then it is provided by docker

Reply:
But you are not wrong, for instance nerdctld would show up as "docker" too...

One could look at Platform (if available), and look for "Docker Engine" there?

Reply:
`!info.Platform?.Name || info.Platform?.Name === '' || info.Platform?.Name?.startsWith('Docker Engine'`

But it does seem a bit paranoid, I don't anyone but Podman that is overwriting the Docker `/var/run/docker.sock`

Reply:
Normally when using a different engine, you would configure the docker client but leave the docker server alone.

Something like DOCKER_HOST: https://github.com/afbjorklund/nerdctld?tab=readme-ov-file#running-daemon